src/xunit.execution/Sdk/Frameworks/Runners/XunitTestClassRunner.cs

```diff
@@ -88,9 +93,12 @@ protected virtual void CreateClassFixture(Type fixtureType)
         async Task CreateClassFixtureAsync(Type fixtureType)
         {
             CreateClassFixture(fixtureType);
-            foreach(var uninitializedFixture in ClassFixtureMappings.Values.OfType<IAsyncLifetime>())
-                await Aggregator.RunAsync(uninitializedFixture.InitializeAsync);
-            
+            foreach (var uninitializedFixture in ClassFixtureMappings.Values.OfType<IAsyncLifetime>().ToList())
+                if (!InitializedAsyncFixtures.Contains(uninitializedFixture))
+                {
+                    InitializedAsyncFixtures.Add(uninitializedFixture);
+                    await Aggregator.RunAsync(uninitializedFixture.InitializeAsync);
```

Comment:
Thank you for the feedback. I have rewritten the function to adopt the changes from #870 and include the fix for repeated initialisation.